src/utils.rs

```diff
@@ -57,3 +57,18 @@ pub fn effective_value(output: &OutputGroup, feerate: f32) -> u64 {
         .value
         .saturating_sub(calculate_fee(output.weight, feerate))
 }
+
+// Returns the weight of the output group(s)
+pub fn calculate_base_weight_btc(output_weight: u64) -> u32 {
+    // VERSION_SIZE: 4 bytes - 16 WU
+    // SEGWIT_MARKER_SIZE: 2 bytes - 2 WU
+    // NUM_INPUTS_SIZE: 1 byte - 4 WU
+    // NUM_OUTPUTS_SIZE: 1 byte - 4 WU
+    // NUM_WITNESS_SIZE: 1 byte - 1 WU
+    // LOCK_TIME_SIZE: 4 bytes - 16 WU
+    // OUTPUT_VALUE_SIZE: variable
+
+    // Total default: (16 + 2 + 4 + 4 + 1 + 16 = 43 WU + variable) WU
+    // Source - https://docs.rs/bitcoin/latest/src/bitcoin/blockdata/transaction.rs.html#599-602
+    output_weight as u32
```

Comment:
you are not adding the default weights here

Reply:
conversion to u32 should not happen.
output_weight should be an u32

Reply:
The weight method from txOut gives a u64
https://docs.rs/bitcoin/latest/src/bitcoin/blockdata/transaction.rs.html#599-602

Reply:
`CoinSelectOpt` is generalized for all blockchains (Bitcoin and Bitcoin adjacent), so it didn't change it. Instead, I converted the output from the function to accommodate our requirements. 

The Base Weight is the Total Weight for the entire output transaction as given in the comment line 606.

Reply:
now you cab proceed

Reply:
done